driver-kafka/src/main/java/io/openmessaging/benchmark/driver/kafka/KafkaBenchmarkConsumer.java

```diff
@@ -51,7 +51,7 @@ public KafkaBenchmarkConsumer(KafkaConsumer<byte[], byte[]> consumer, ConsumerCa
 
                 Map<TopicPartition, OffsetAndMetadata> offsetMap = new HashMap<>();
                 for (ConsumerRecord<byte[], byte[]> record : records) {
-                    callback.messageReceived(record.value());
+                    callback.messageReceived(record.value(), System.nanoTime());
```

Comment:
Unfortunately despite all these changes, I'm getting consumer rate of zero, so all messages are "missed"; I'm trying to troubleshoot this, but I might need your help @merlimat in figuring out how consumer get invoke here. As I mentioned, launching consumer in constructor has been problematic for me.